### DISCUSSION TITLE
Test build for word search

### DISCUSSION THREAD

**Discussion by TBog** [TBog/TBLauncher#255] [Pre-Release]
The searched text is split by the ' '(space) character into words. An entry must match all the words in order to be a result.

<hr /><em>This discussion was created from the release <a href='https://github.com/TBog/TBLauncher/releases/tag/v6.10-feature_multi_search'>Test build for word search</a>.</em>

**Comment by victor0d** [comment]
I think this would be excellent.

And actually maybe only two words need to be considered. i.e. only one ' ' (space), ignoring any space after the first one. This would allow matching tag+tag or tag+name.

I am not sure if extending the search to three or even more would have any performance impact.

**Reply by TBog** [reply]
A test build is already available.
The search is done for each word. It's a recursive algorithm so any number of words can be used.
Search speed should not be a problem.

**Comment by victor0d** [comment]
Thanks for pointing it out.
I have just tested it: 1/2/3/4 words,  swapping the tags order, names, ... Flawlessly.
AMAZING. Specially considering that it is a test build. Feels like an stable release.

**Reply by TBog** [reply]
It's a test for other things that may be necessary for this feature.
* I noticed I'm missing a preference toggle for this. Is it necessary or not?
* Should the word separator be customizable? Replace space with dot or maybe some other languages have a nbsp character. Maybe have a list of separator characters?
* An option to have the first word require a full tag match.

**Comment by victor0d** [comment]
I understand.

I can see that the feature **could** have some interference with the web search. This is only case that I can think of right now, where spaces are used. But I have just been testing it, and I hadn't feel any interference at all when typing text to carry web searches.

Considering this, and with regards of your questions:
- A **customizable separator** is always useful. Although I can't think on many cases when this will be used. I guess this could depend on the keyboard used. Personally I am happy with the space.
- A **feature toggle**, could be easily generated by the user selecting an almost impossible character as separator.
- A **first full match** is a really fancy toggle :o). Not so sure about the usage, because it means much more typing. But maybe I am wrong here.
